### PR TITLE
[WIP] - Add stack configuration files

### PR DESCRIPTION
**Pull request by bollu**
This will allow new users to compile the library using `stack build`, run examples using `stack exec`, etc.
### TODO
- [ ] Update `README.md`
- [ ] generate executables for all examples

### CODE REVIEW

File: plots.cabal [hunk 1]
```diff
@@ -87,3 +87,26 @@ library
   ghc-options:         -Wall
 
   default-language:  Haskell2010
+
+-- are plot example
```
Comment (bergey) [inline plots.cabal]: should that say `area plot`?

File: stack.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# For more information, see: https://github.com/commercialhaskell/stack/blob/master/doc/yaml_configuration.md
+
+# Specifies the GHC version and set of packages available (e.g., lts-3.5, nightly-2015-09-21, ghc-7.10.2)
+resolver: lts-3.2
```
Comment (bergey) [inline stack.yaml:4]: Why not lts-5?

**Comment by cchalmers** [comment]
Thanks for working on this.

How do you plan on running all the examples? It looks like you've added an individual example to cabal but I'm not sure it's a good idea to do this for so many examples. Maybe you could make a script that runs all the examples in the examples folder.

I've made some of the examples use the command line interface. So you'd run them using something like

```
stack runghc -- examples/area.hs -o examples/area.png -w500
```

Also note that some of the examples use extra packages (like [`wreq`](http://hackage.haskell.org/package/wreq)).

File: plots.cabal (hunk 1 above)
Comment (bollu) [inline plots.cabal]: yep, sorry! I was just trying to get a quick-and-dirty PR against the branch that I can iterate on

**Comment by bollu** [comment]
Yes, I was thinking of generating all the executables one by one. I think doing it by hand is the cleanest solution (but your opinion may differ). 

Running the examples could be scripted, but I think the builds should still happen through stack.

**Comment by bollu** [comment]
Also, I think you can add a 

```
#!/usr/bin/env runhaskell
```

to the top of haskell files to have them run automatically with a `./file-name.hs`

**Comment by bollu** [comment]
@cchalmers opinions on the build system for examples?

**Comment by cchalmers** [comment]
I like the idea of having something on each file. I've added some config and comments to the [stocks](https://github.com/cchalmers/plots/blob/master/examples/stocks.hs) example. The stack config also lets you add extra packages which is nice.

And then maybe a script than builds all the examples in the examples folder. The script can be in cabal, I just don't want 30 or so examples in the cabal file.

**Comment by bollu** [comment]
that's fair I guess. What do you think of using `shake` as a build system for plots to build the examples? would that be overkill? (I guess I'm looking for excuses to over-engineer here).

I didn't know that you could script in cabal, do you mean using cabal-the-library to compile the code?

**Comment by cchalmers** [comment]
I agree that `shake`'s probably overkill but use it if you want. :)

I meant add an option to cabal that would run the script that runs the examples but I'm not sure it's actually a thing.

**Comment by bollu** [comment]
Hm, I'll take a look and decide :)

**Comment by bollu** [comment]
I asked on the IRC, and `lyxia` commented:

```
it would be much simpler to expect a user to type 'stack runghc example.hs' themselves :)
```

which I do agree with tbh - seems cleaner to teach the users on how to run the examples, doesn't it?

**Comment by cchalmers** [comment]
Yeah, I'm happy with that.

File: stack.yaml (hunk 2 above)
Comment (bollu) [inline stack.yaml:4]: the default one is `lts-3.2`. Will change to `lts-5`

**Comment by sid-kap** [comment]
@bollu Can we close this, since #9 accomplished some of the goals of this PR?

**Comment by bollu** [comment]
yep, sure :). Sorry for not being around - I was busy with other stuff!

**Pull request closed without merging**